src/engine/world/actor/player/player.ts

```diff
@@ -831,31 +849,36 @@ export class Player extends Actor {
         return this.equipment.items[equipmentIndex(equipmentSlot)] || null;
     }
 
-    public canEquipItem(item: ItemDetails): boolean {
-        const requirements = item.equipmentData?.requirements;
-        if (!requirements) return true;
-
-        const hasSkillRequirements = Object.entries(requirements.skills || {}).every(([skill, level]) => this.skills.hasLevel(skill as SkillName, level));
-        const hasQuestRequirements = Object.entries(requirements.quests || {}).every(([quest, stage]) => this.getQuest(quest).progress >= stage);
 
-        return hasSkillRequirements && hasQuestRequirements;
-    }
-
-    public missingItemEquipRequirements(item: ItemDetails): string[] {
+    /**
+     * Check if a player can equip an item
+     * @param item either an ItemDetails instance or the string id of the item to be checked
+     * @return {equipable: boolean, missingRequirements: string[]} equipable is false if for any reason the item can not
+     * be equipped, if it can not be equipped, a list of reasons are attached as the missingRequirements array
+     *
+     * defaults to equipable=true if the item string id does not exist
+     */
+    public canEquipItem(item: ItemDetails | string): { equipable: boolean, missingRequirements?: string[] } {
+        if(typeof item === 'string') {
+            item = itemMap[item];
+            if(!item) {
+                return { equipable: true }
+            }
+        }
         const missingRequirements = [];
         const requirements = item.equipmentData?.requirements;
-        if (!requirements) return missingRequirements;
+        if (!requirements) return { equipable: true };
 
         missingRequirements.push(
             ...Object.entries(requirements.skills || {})
                 .filter(([skill, level]) => !this.skills.hasLevel(skill as SkillName, level))
                 .map(([skill, level]) => `You need to be at least level ${level} ${skill} to equip this item.`),
             ...Object.entries(requirements.quests || {})
-                .filter(([quest, stage]) => this.getQuest(quest).progress < stage)
-                .map(([quest]) => `You must progress further in the ${quest.replace(/_/g, ' ')} quest to equip this item.`)
+                .filter(([quest, stage]) => this.hasQuestRequirement(quest, stage))
+                .map(([quest]) => `You must progress further in the ${quest.replace(/^([a-z]+:)/gm, '').replace(/_/g, ' ')} quest to equip this item.`)
         );
 
-        return missingRequirements;
+        return { equipable: true, missingRequirements: missingRequirements };
```

Comment:
Should this be checking if there is any missing requirements rather than always returning true?
```suggestion
        return { equipable: missingRequirements.length === 0, missingRequirements: missingRequirements };
```